### PR TITLE
Gershayim

### PR DESCRIPTION
**Pull request by YishaiGlasner**
should fix this issue - https://app.shortcut.com/sefaria/story/760/exact-term-not-recognized-in-search - and similar issues (for example רלב״ג קהלת which is ref and not author).
in fact there are 2 issues:
1. the auto completion - that's fixed on server side.
2. the search itself (when pressing enter) - that's fixed on client side.

### CODE REVIEW

File: sefaria/model/autospell.py [hunk 1]
```diff
@@ -36,7 +36,7 @@
 
 def normalizer(lang):
     if lang == "he":
-        return lambda x: "".join([c if c in letter_scope else unidecode(c) for c in hebrew.normalize_final_letters_in_str(x)])
+        return lambda x: "".join([c if c in letter_scope else unidecode(c) for c in hebrew.normalize_final_letters_in_str(x)]).replace('"', '״')
```
Comment (nsantacruz) [inline sefaria/model/autospell.py:39]: unidecode will convert gershayim to ASCII quotes. We're already using it here. All we need to do is remove gershayim from `letter_scope`.

See this:
```python
In [3]: from unidecode import unidecode

In [4]: unidecode('״')
Out[4]: '"'

```

Comment (nsantacruz) [reply sefaria/model/autospell.py:39]: Note, in your code, the standard quotation mark is gershayim and ASCII is converted to gershayim. If we use unidecode, the standard quotation mark will be ASCII.

File: static/js/Header.jsx [hunk 2]
```diff
@@ -310,6 +310,14 @@ class SearchBar extends Component {
           this.submitSearch(Sefaria.repairCaseVariant(query, d));
           return;
         }
+        if (Sefaria.isGershayimVariant(query, d)) {
+          if (query.includes('"')) {
```
Comment (nsantacruz) [inline static/js/Header.jsx:314]: Why is this `if` necessary? We already know it's a gershayim variant. Shouldn't we just replace all gershayim with ASCII quotes?

File: static/js/sefaria/sefaria.js [hunk 3]
```diff
@@ -1943,6 +1943,17 @@ _media: {},
...
           data["completions"][0] != query.slice(0, data["completions"][0].length))
   },
+  isGershayimVariant: function(query, data) {
+    function replace_gershayim_by_merkhaot(string) {
+        return string.replace('"', '״');
+    }
+    return (
+      !(data["is_ref"]) &&
+      data.completions &&
+      !data.completions.includes(query) &&
```
Comment (nsantacruz) [inline static/js/sefaria/sefaria.js:1953]: previous line is redundant. this line can be `!data.completions?.includes(query)`

File: static/js/sefaria/sefaria.js [hunk 4]
```diff
@@ -1943,6 +1943,17 @@ _media: {},
...
+      data.completions.map(x => replace_gershayim_by_merkhaot(x)).includes(replace_gershayim_by_merkhaot(query))
```
Comment (nsantacruz) [inline static/js/sefaria/sefaria.js:1954]: I don't think `map()` will work in this case. It returns an array but you want a boolean. You're also checking `includes()` but don't you want to check for exact equality?

File: static/js/sefaria/sefaria.js [hunk 5]
```diff
@@ -1943,6 +1943,17 @@ _media: {},
           data["completions"][0].toLowerCase().replace('״','"') == query.slice(0, data["completions"][0].length).toLowerCase().replace('״','"') &&
           data["completions"][0] != query.slice(0, data["completions"][0].length))
   },
+  isGershayimVariant: function(query, data) {
+    function replace_gershayim_by_merkhaot(string) {
```
Comment (nsantacruz) [inline static/js/sefaria/sefaria.js:1947]: I don't understand this function name. Should it be `replace_quotes_with_gershayim()`? (also, note the order of the replace will need to change if we switch to using unidecode).

File: static/js/Header.jsx (hunk 2 above)
Comment (YishaiGlasner) [reply static/js/Header.jsx:314]: how do we know that?

File: static/js/sefaria/sefaria.js (hunk 4 above)
Comment (YishaiGlasner) [reply static/js/sefaria/sefaria.js:1954]: we're replacing all the gershayim in all completions, and checking if the new array includes the quiry string (also after replacing)

File: static/js/sefaria/sefaria.js (hunk 5 above)
Comment (YishaiGlasner) [reply static/js/sefaria/sefaria.js:1947]: the order doesn't matter. we're not changing the server response. just how it compares query to data (i'll change the name)

File: static/js/sefaria/sefaria.js (hunk 4 above)
Comment (YishaiGlasner) [reply static/js/sefaria/sefaria.js:1954]: but using 'some' will be better

Comment (nsantacruz) [reply static/js/sefaria/sefaria.js:1954]: ah! I misunderstood this. That makes sense.

File: static/js/Header.jsx (hunk 2 above)
Comment (nsantacruz) [reply static/js/Header.jsx:314]: Ah! now I understand the issue. Can you wrap this if statement in a function similar to `repairCaseVariant`?

File: static/js/Header.jsx [hunk 6]
```diff
@@ -310,12 +310,10 @@ class SearchBar extends Component {
...
         }
-        if (Sefaria.isGershayimVariant(query, d)) {
-          if (query.includes('"')) {
-            this.submitSearch(query.replace('"', '״'));
-          } else {
-            this.submitSearch(query.replace('״', '"'));
-          }
+        const repairedQuery = Sefaria.repairGershayimVariant(query, d);
+        if (repairedQuery !== query) {
+          console.log(query, d);
```
Comment (nsantacruz) [inline static/js/Header.jsx:315]: Can you remove this console.log?

**Review by nsantacruz — DISMISSED** [review]

**Review by nsantacruz — APPROVED** [review]

**Pull request merged**